wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/interfaces/EntityDocument.java

```diff
@@ -46,5 +46,10 @@
 	 * @return revision id
 	 */
 	long getRevisionId();
+	
+	/**
+	 * Returns a copy of this document with an updated revision id.
+	 */
+	EntityDocument withRevisionId(long newRevisionId);
```

Comment:
What is the use case of this setter? It seems quite dangerous to allow setting the revisionId outside of the deserialization of content retrieved from Wikibase. It is maybe useful when building the result of an edit action but we should at least add a strong warning to this method.

Reply:
Yes - it is not meant to be used by library users, but I haven't found a better solution. Here is the use case:
* you have an EntityDocument
* you edit some part of it (add an alias, say) with the API
* the server returns the new revid and the new list of aliases for that language
* we need to construct the new version of the document, which is: the initial state of the document, with the updated aliases and the updated revision id. Hence `withRevisionId` and `withAliases`.

One thing we could do is merge the two methods: each `with` method of `EntityDocument`, `StatementDocument`, etc would also take the new revision id as a second parameter. This way, there would be no natural way to change just the revision id, or just the content.

Reply:
Ok, I see.

> One thing we could do is merge the two methods: each with method of EntityDocument, StatementDocument, etc would also take the new revision id as a second parameter. 

It looks quite heavy but why not.

An other question: do we want to keep the revision id if the entity is modified without new revision id given? If we remove it it would have the advantage of not having entity objects in a state that is not the one of their revision id. But it would require the toolkit user to give the base revision id when trying to save an edited entity.

Reply:
so you would silently "reset" the revision id in `withDescription` for instance? users might not expect that…

if we go for `withDescription(desc, newRevId)` then I believe we can get rid of `TermedStatementDocument` altogether (because there will be no intersection between the methods of `StatementDocument` and `TermedDocument`).